tools/chplcheck/src/driver.py

```diff
@@ -120,13 +121,19 @@ def basic_rule(self, pat):
 
         The name of the decorated function is used as the name of the rule.
         """
-
-        def wrapper(func):
+        def decorator_basic_rule(func):
             self.BasicRules.append((func.__name__, pat, func))
-            return func
-        return wrapper
+            if not default:
+                self.SilencedRules.append(func.__name__)
+
+            @functools.wraps(func)
+            def wrapper_basic_rule(*args, **kwargs):
+                return func(*args, **kwargs)
+            return wrapper_basic_rule
```

Comment:
In this situation, could you not return `func`?

Reply:
Yes, technically you can just return `func` and then get rid of 1 layer of wrapper. But then the rule can't be called recursively